Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -184,25 +184,62 @@ namespace
         }
     }
 
+    void SetXRGamepadObjectData(Napi::Object& jsInputSource, Napi::Object& jsGamepadObject, xr::System::Session::Frame::InputSource& inputSource)    
+    {
+        auto env = jsInputSource.Env();
+        //Set Gamepad Object
+
+        auto gamepadButtons = Napi::Array::New(env, inputSource.GamepadObject.Buttons.size());
+        for (size_t i = 0; i < inputSource.GamepadObject.Buttons.size(); i++)
+        {
+            auto gamepadButton = Napi::Object::New(env);
+            auto napiGamepadPressed = Napi::Boolean::New(env, inputSource.GamepadObject.Buttons[i].Pressed);
+            auto napiGamepadTouched = Napi::Boolean::New(env, inputSource.GamepadObject.Buttons[i].Touched);
+            auto napiGamepadValue = Napi::Number::New(env, inputSource.GamepadObject.Buttons[i].Value);
+            gamepadButton.Set("pressed", napiGamepadPressed);
+            gamepadButton.Set("touched", napiGamepadTouched);
+            gamepadButton.Set("value", napiGamepadValue);
+            gamepadButtons.Set(static_cast<int>(i), gamepadButton);
+        }
+        jsGamepadObject.Set("buttons", gamepadButtons);
+
+        auto gamepadAxes = Napi::Array::New(env, inputSource.GamepadObject.Axes.size());
+        for (size_t i = 0; i < inputSource.GamepadObject.Axes.size(); i++)
+        {
+            auto napiGamepadAxesValue = Napi::Number::New(env, inputSource.GamepadObject.Axes[i]);
+            gamepadAxes.Set(static_cast<int>(i), napiGamepadAxesValue);
+        }
+        jsGamepadObject.Set("axes", gamepadAxes);
+        jsInputSource.Set("gamepad", jsGamepadObject);
+    }
+
     Napi::ObjectReference CreateXRInputSource(xr::System::Session::Frame::InputSource& inputSource, Napi::Env& env)
     {
         constexpr std::array<const char*, 2> HANDEDNESS_STRINGS{
             "left",
             "right"};
         constexpr const char* TARGET_RAY_MODE{"tracked-pointer"};
-
         auto jsInputSource = Napi::Object::New(env);
+
         jsInputSource.Set("handedness", Napi::String::New(env, HANDEDNESS_STRINGS[static_cast<size_t>(inputSource.Handedness)]));
         jsInputSource.Set("targetRayMode", TARGET_RAY_MODE);
         SetXRInputSourceData(jsInputSource, inputSource);
 
         auto profiles = Napi::Array::New(env, 1);
-        Napi::Value string = Napi::String::New(env, "generic-trigger-squeeze-touchpad-thumbstick");
+        Napi::Value string = Napi::String::New(env, "windows-mixed-reality");
```

Comment:
My concern here is in line with some of my other comments, but it's more emphatic here: this cannot be in this file. The `generic-trigger-squeeze-touchpad-thumbstick` string was already probably not as it should be, but this file definitely cannot be making decisions that target a specific platform. I think there might already be a way to provide this string, or something very much like it, to Babylon.js's XR abstraction; if you could leverage that to designate the preferred controller type and just use that string here, that would probably be ideal. If that's too much of a change to be a part of this PR and it's more convenient to have this hard-coded in the C++ for now, it's probably alright to have this in a header and use one of our CMake platform tricks to have it only take this new value on Windows (and have it be either generic or undefined on other platforms). I don't think it can go here, though. NativeXr.cpp exists to bridge the gap between Babylon.js and XR.h; it _cannot_ know about Windows.